web-platform/website/src/components/pages/account-settings/EditAccountEmailView.tsx

```diff
@@ -0,0 +1,278 @@
+    validationResult.password.error
+  );
+
+  // Prevents closure in validateAndUpdate
+  const emailRef = useRef(email);
+  emailRef.current = email;
+
+  const passwordRef = useRef(password);
+  passwordRef.current = password;
+
+  useEffect(() => {
+    setValidationResult((validationResult) => ({
+      ...(validationResult || {
+        email: { error: undefined },
+        password: { error: undefined },
+      }),
+      valid: true,
+      passwordRules: [],
+    }));
+  }, []);
+
+  const { changeUserEmail } = useAuthentication();
+
+  const handleChange = (e: any) => {
+    setValidationResult({
+      email: {},
+      password: {},
+      passwordRules: [],
+      valid: true,
+    });
+
+    setDisabled(false)
+    if (e.target.value === "") {
+      setDisabled(true)
+    }
+    if (e.target.name === "password") {
+      setPassword(e.target.value)
+      setVisibility(true)
+      return
+    }
+    if (e.target.name === "email") {
+      setEmail(e.target.value)
+      return
+    }
+  };
+
+
+  async function validateAndUpdate() {
+    setValidationResult(undefined);
+
+    const validationResult = validateLoginForm(
+      emailRef.current,
+      passwordRef.current
+    );
+
+    setValidationResult(validationResult);
+
+    try {
+      await changeUserEmail(emailRef.current, passwordRef.current);
+      setConfirmationView(true)
+    } catch (e) {
+
+      const error = getFirebaseErrorMessage(e as FirebaseError)
+      let emailErr = ""
+      let passwordErr = ""
+
+      error.indexOf(firebaseStings["auth/email-already-in-use"]) > -1 || error.indexOf(firebaseStings["auth/invalid-email"]) > -1 ? emailErr = error : emailErr = ""
+
+      error.indexOf(firebaseStings["auth/wrong-password"]) > -1 ? passwordErr = error : passwordErr = ""
+
+      setValidationResult({
+        email: { error: emailErr },
+        password: { error: passwordErr },
+        passwordRules: [],
+        valid: false,
+      });
+
+      setVisibility(false)
+    }
+  }
+
+  return (
+    <Card className="flex-nowrap p-4">
+      {confirmationView ? (
+        <EmailChangedView email={emailRef.current} />
+      )
+        :
+        (
+          <Container
+            className={`${ContainerStyles.NoSpacing} ${styles.container} p-0`}
+          >
+            <Row className="mb-3">
+              <Col>
+                <h1 className={Fonts.Headline}>{strings.title}</h1>
+              </Col>
+            </Row>
+            <Row className="mb-3">
+              <Col>
+                <Form>
+                  <FormGroup className="mb-4">
+                    <Label for="email" className="fw-bold">
+                      {strings.newEmail}
+                    </Label>
+                    <Input
+                      id="email"
+                      type="email"
+                      name="email"
+                      autoFocus={true}
+                      value={email}
+                      onChange={handleChange}
```

Comment:
(e) => setEmail(e.target.value). Same for other fields.

Reply:
Thank you!